pkg/apis/autopilot/v1alpha1/autopilotrule.go

```diff
@@ -100,6 +99,51 @@ type AutopilotRuleSpec struct {
 	ActionsCoolDownPeriod int64 `json:"actionsCoolDownPeriod,omitempty"`
 }
 
+// AutopilotRuleStatus captures the status of various objects that are monitored as part of an autopilot rule
+type AutopilotRuleStatus struct {
+	// Objects is a map of objects that are monitored as part of a autopilot rule and their statuses
+	Objects map[RuleStatusObjectKey][]*AutopilotRuleObjectStatus `json:"objects"`
```

Comment:
I am a bit worried we will end up creating a huge autopilot object in Kubernetes etcd, with all the objects in it. For ex, if autopilot is monitoring 1000PVCs, they will end up in this single object. ATM, I am not sure what the consequences will be. 
What is the downside of creating an AutopilotObject or just updating statuses on existing/upcoming objects CRDs

Reply:
Good point. In the actual autopilot PR that updates the status, I'm doing 2 levels of trimming.

1. Keeping only last 10 statutes per object
2. Keeping statuses only over last 48 hours. We can drop this retention time down for large scale clusters

This will help to keep the object size in check.

About creating AutopilotObject vs updating status here,

- The scale issue will remain. It's one large object vs many small objects
- Users applied the autopilot rule so that's where they would want to see the status. So if we add a new CRD called AutopilotObject, we will have to add the co-relation.

Reply:
Lets do this exercise:
- Find the base size of the status object and autopilot rule
- With message size limited to 100 bytes (100 chars) and name limited to 20 bytes (20 chars)
- Lets calculate the total no. of status objects we can have that can fit in 1.5MB
With this we can have a ballpark number of how many objects, autopilot can sustain at any point of time.

Reply:
Decoupled status from the rule and add it's own object